web-registry/src/components/organisms/ChooseCreditClass/ClassOptions.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Grid } from '@mui/material';
+
+import { ChooseClassTemplate } from './ChooseClassTemplate';
+import { ClassOption } from './ClassOption';
+import { useGetCreditClassOptions } from './hooks/useGetCreditClassOptions';
+import { validate, getProjectPageBaseData } from '../../../lib/rdf';
+import {
+  useUpdateProjectByIdMutation,
+  useShaclGraphByUriQuery,
+} from '../../../generated/graphql';
+
+const ClassOptions: React.FC = () => {
+  const [error, setError] = useState('');
+  const navigate = useNavigate();
+  const { projectId } = useParams();
+  const { creditClassOptions, loading } = useGetCreditClassOptions();
+  const { data: graphData } = useShaclGraphByUriQuery({
+    variables: {
+      uri: 'http://regen.network/ProjectPageShape',
+    },
+  });
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  // Reset error message, so it reappears when user tries again
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => {
+        setError('');
+      }, 5000);
+    }
+  }, [error]);
+
+  async function handleSelection(
+    creditClassId: string,
+    creditClassOnChainId: string,
+  ): Promise<void> {
+    if (graphData?.shaclGraphByUri?.graph) {
+      const metadata = getProjectPageBaseData(creditClassOnChainId);
+
+      try {
+        const report = await validate(
+          graphData.shaclGraphByUri.graph,
+          metadata,
+          'http://regen.network/ProjectPageCreditClassGroup',
+        );
+        if (report.conforms) {
+          try {
+            await updateProject({
+              variables: {
+                input: {
+                  id: projectId,
+                  projectPatch: {
+                    creditClassId,
+                    metadata,
+                  },
+                },
+              },
+            });
+
+            navigate(`/project-pages/${projectId}/basic-info`);
+          } catch (e) {
+            setError(`There was a problem updating this project: ${e}`);
+          }
+        }
+      } catch (e) {
+        setError(`There was a problem validating this credit class: ${e}`);
+      }
+    }
+  }
+
+  return (
+    <ChooseClassTemplate
+      justifyContent={creditClassOptions?.length > 1 ? 'flex-start' : 'center'}
+      loading={loading}
+      error={error}
+    >
+      {creditClassOptions?.length > 0 ? (
+        creditClassOptions?.map(c => (
+          <ClassOption
+            key={c.onChainId}
+            title={c.title}
+            imgSrc={c.imageSrc}
+            description={c.description}
+            onClick={() => handleSelection(c.id, c.onChainId)}
```

Comment:
Just to make it a bit more explicit, or we could destruct directly.

```suggestion
        creditClassOptions?.map(classOption => (
          <ClassOption
            key={classOption.onChainId}
            title={classOption.title}
            imgSrc={classOption.imageSrc}
            description={classOption.description}
            onClick={() => handleSelection(classOption.id, classOption.onChainId)}
```